pandas/errors/__init__.py

```diff
@@ -38,9 +38,53 @@ class ParserError(ValueError):
 
 class DtypeWarning(Warning):
     """
-    Warning that is raised for a dtype incompatibility. This
-    can happen whenever `pd.read_csv` encounters non-
-    uniform dtypes in a column(s) of a given CSV file.
+    Warning raised when reading different dtypes in a column from a file.
+
+    Raised for a dtype incompatibility. This can happen whenever `read_csv`
+    or `read_table` encounter non-uniform dtypes in a column(s) of a given
+    CSV file.
+
+    See Also
+    --------
+    pandas.read_csv : Read CSV (comma-separated) file into a DataFrame.
+    pandas.read_table : Read general delimited file into a DataFrame.
+
+    Notes
+    -----
+    This warning is issued when dealing with larger files because the dtype
+    checking happens per chunk read.
+
+    Despite the warning, the CSV file is read with mixed types in a single
+    column which will be an object type. See the examples below to better
+    understand this issue.
+
+    Examples
+    --------
+    This example creates and reads a large CSV file with a column that contains
+    `int` and `str`.
+
+    >>> df = pd.DataFrame({'a':['1']*100000 + ['X']*100000 + ['1']*100000,
+    ...                    'b':['b']*300000})
+    >>> df.to_csv('test.csv', index=False)
+    >>> df2 = pd.read_csv('test.csv')
+    >>> DtypeWarning: Columns (0) have mixed types... # doctest: +SKIP
```

Comment:
Can you remove the `>>>` on this line? (it might be you need to move the "# doctest: +SKIP" to the line above after the read_csv)

Reply:
@jorisvandenbossche @TomAugspurger I've added "import os" and an "os.remove('test.csv')" at the end of the example, just as you've agreed on #20302 , and also put "# doctest: +SKIP" right before the warning, like this: 

    >>> import os
    >>> df = pd.DataFrame({'a':['1']*100000 + ['X']*100000 + ['1']*100000,
    ...                    'b':['b']*300000})
    >>> df.to_csv('test.csv', index=False)
    >>> df2 = pd.read_csv('test.csv')
    >>> os.remove('test.csv')
    # doctest: +SKIP
    DtypeWarning: Columns (0) have mixed types...

But I got this message on validation:

################################################################################
################################### Doctests ###################################
################################################################################

**********************************************************************
Line 32, in pandas.errors.DtypeWarning
Failed example:
    os.remove('test.csv')
Expected:
    # doctest: +SKIP
    DtypeWarning: Columns (0) have mixed types...
Got nothing


And also, removing ".csv" from to_csv and read_csv, like Tom suggested, raises two possible errors:
>>> os.remove('test.csv'): FileNotFoundError: [Errno 2] No such file or directory: 'test.csv'
>>> os.remove('test'): FileNotFoundError: File b'test' does not exist

What do you guys recommend?